### PR TITLE
Eco 1047/tests

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: scripts/src/client.ts [hunk 1]
```diff
@@ -320,18 +319,22 @@ export class Client {
 		}
 	}
 
-	public async submitOrder(orderId: string, content?: string): Promise<Order> {
+	public async submitOrder(orderId: string, options: {content?: string, transaction?: string} = {}): Promise<Order> {
```
Comment (nitzantomer) [inline scripts/src/client.ts]: i'd do it like this:
```
public async submitOrder(orderId: string, options: { content: string } = {}): Promise<Order>;
public async submitOrder(orderId: string, options: { transaction: string } = {}): Promise<Order>;
public async submitOrder(orderId: string, options: { content?: string, transaction?: string } = {}): Promise<Order> {
    ...
}
```
that way when using this it's clear that i must pass either `{ content: "..." }` or `{ transaction: "..." }`

File: scripts/src/client.ts [hunk 2]
```diff
@@ -320,18 +319,22 @@ export class Client {
...
 		try {
 			const res = await this.requests
-				.request(`/v2/orders/${ orderId }`, { content })
+				.request(`/v2/orders/${ orderId }`, { transaction: options.transaction, content: options.content })
```
Comment (nitzantomer) [inline scripts/src/client.ts]: why not just:
```
.request(`/v2/orders/${ orderId }`, options)
```
?

File: scripts/src/client.ts [hunk 3]
```diff
@@ -320,18 +319,22 @@ export class Client {
...
 				.post<Order>();
 			return res.data;
 		} catch (e) {
-			console.log(`error while submitting order ${ orderId } with content: "${ content }"`);
+			console.log(`error while submitting order ${ orderId } with options: "${ options }"`);
```
Comment (nitzantomer) [inline scripts/src/client.ts]: that will stringify the `options`, this should be better:
```
console.log("error while submitting order ${ orderId } with options: ", options);
```

File: scripts/src/client.ts [hunk 4]
```diff
@@ -371,7 +374,7 @@ export class Client {
 			throw new Error("first set a wallet");
 		}
 
-		await this.wallet.trustKin();
+		// NOP await this.wallet.trustKin();
```
Comment (nitzantomer) [inline scripts/src/client.ts]: just remove it, no?

File: scripts/src/client.ts [hunk 5]
```diff
@@ -62,27 +58,44 @@ export class ClientError extends Error {
...
 
-	public static async getConfig(): Promise<ConfigResponse> {
+	public static async getServerConfig(): Promise<ConfigResponse> {
 		const res = await axios.get<ConfigResponse>(MARKETPLACE_BASE + "/v2/config");
 		return res.data;
 	}
 
+	private static extractAppId(jwt: string) {
+		return (jsonwebtoken.decode(jwt) as { iss: string }).iss;
+
```
Comment (liorama) [inline scripts/src/client.ts]: Extra line is extra

File: scripts/src/client.ts [hunk 6]
```diff
@@ -62,27 +58,44 @@ export class ClientError extends Error {
...
 export class ClientRequests {
-	public static async create(data: any, headers?: StringMap) {
-		const res = await axios.post<{ auth: AuthToken; }>(MARKETPLACE_BASE + "/v2/users", data, { headers });
-		return new ClientRequests(res.data.auth);
+	public static async create(data: SignInPayload, headers?: StringMap) {
+		// get blockchain version for current app
+		const appId = this.extractAppId(data.jwt);
+		const blockchainVersion = (await axios.get<number>(
+				MARKETPLACE_BASE + `/v2/applications/${ appId }/blockchain_version`)
+		).data.toString() as BlockchainVersion;
```
Comment (liorama) [inline scripts/src/client.ts]: Thid id real ugly, can you extract that the url into a const

File: scripts/src/client.ts [hunk 7]
```diff
@@ -210,11 +208,12 @@ export class Client {
...
 
 	public async updateWallet(walletAddress?: string) {
+		const kinjs = this.requests.blockchainVersion === "2" ? kinjs1 : kinjs2;
 		const keys = !walletAddress ?
-			Keypair.random() :
+			kinjs.Keypair.random() :
 			(walletAddress.startsWith("S") ?
-				Keypair.fromSecret(walletAddress) :
-				Keypair.fromPublicKey(walletAddress));
+				kinjs.Keypair.fromSecret(walletAddress) :
```
Comment (liorama) [inline scripts/src/client.ts]: How is `walletAddress` a secret?

File: scripts/src/client.ts (hunk 5 above)
Comment (liorama) [inline scripts/src/client.ts]: Extra line is extra

File: scripts/src/client.ts (hunk 6 above)
Comment (liorama) [inline scripts/src/client.ts]: These 3 lines are butt ugly, can you extact the url string into a const please

Comment (doodyparizada) [reply scripts/src/client.ts]: I don't see why extracting the URL to a consts makes it nicer. The rest of the code uses inline urls when making requests. The second part of `.data.toString()` can be moved/ turned into a function maybe

File: scripts/src/client.ts (hunk 7 above)
Comment (doodyparizada) [reply scripts/src/client.ts]: when `walletAddress.startsWith("S")`

File: scripts/src/client.ts (hunk 2 above)
Comment (doodyparizada) [reply scripts/src/client.ts]: right!

**Pull request merged**